arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,199 @@
+        # enabled). The second section is latency statistics for each of the same
+        # columns as the histogram section. The third section is collapsed totals of
+        # the statistical data from all CPUs.
+
+        # For the first and second sections, we want to retain the row/columnt format
+        # in the output for easy recreation of the data as a table. For the third
+        # section, we want to return key:value pairs for each of the IRQ, Thr, and Usr
+        # total stats.
+
+        # $ sudo rtla timerlat hist -u -d 1 -c 1,2 -E 10 -b 10
+        # # RTLA timerlat histogram
+        # # Time unit is microseconds (us)
+        # # Duration:   0 00:00:02
+        # Index   IRQ-001   Thr-001   Usr-001   IRQ-002   Thr-002   Usr-002
+        # 0          1000       944       134       999       998       986
+        # 10            0        55       863         0         1        13
+        # 20            0         1         3         0         0         0
+        # over:         0         0         0         0         0         0
+        # count:     1000      1000      1000       999       999       999
+        # min:          0         3         4         0         2         3
+        # avg:          1         7        10         0         2         3
+        # max:          7        21        24         3        11        15
+        # ALL:        IRQ       Thr       Usr
+        # count:     1999      1999      1999
+        # min:          0         2         3
+        # avg:          0         4         6
+        # max:          7        21        24
+
+        time_unit = ""
+        col_headers = []
+        latency_hist = []
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+        stats_names = ["over:", "count:", "min:", "avg:", "max:"]
+        stats_per_col = []
+        found_all = False
+
+        re_isunit = re.compile(r"^#\ Time\ unit")
+        re_isindex = re.compile(r"^Index")
+        re_isdigit = re.compile(r"^\d")
+        re_isall = re.compile(r"^ALL")
+
+        for line in output.splitlines():
+            if re_isunit.match(line):
+                time_unit = line.split()[4]
+            # Capture the column headers
+            elif re_isindex.match(line):
+                col_headers = line.lower().split()
+            # Stats names repeat, so flag when have passed ^ALL
+            elif re_isall.match(line) and not found_all:
+                found_all = True
+            # Either this is a histogram bucket row, or the first time we have seen
+            # a row beginning with a stat name
+            elif (re_isdigit.match(line)) or (
+                line.split()[0] in stats_names and not found_all
+            ):
+                # Capture the columnar data
+                row_obj = {}
+                for i, col in enumerate(col_headers):
+                    row_obj[col] = line.split()[i]
```

Comment:
This looks like a nice spot for a dictionary comprehension:
```suggestion
                row_obj = {col: line.split()[i] for i, col in enumerate(col_headers)}
```

but, what I think you really want is to use `zip()`:
```suggestion
                row_obj = dict(zip(col_headers, line.split()))
```

Reply:
/me opens his python textbook

Getting academic here, the dict comprehension is concise, for sure, and perhaps shows off more advanced python skills, but I'm curious if it accomplishes anything more than that. Is the resulting bytecode even affected by this syntax difference? IMO while the original is elementary, it is simpler to grok.

While the zip is also more advanced and harder to grok, I do get the point of it.